Add BBEs for GraphQL server-side Caching (#5188)
ballerina-platform/ballerina-distribution: examples/graphql-cache-invalidation/graphql_cache_invalidation.md

> # GraphQL service - Cache Invalidation

Reviewer: ```suggestion
# GraphQL service - Cache invalidation
```

> The Ballerina `graphql` module provides functionality for cache invalidation. The `invalidate()` and `invalidateAll()` APIs in the `graphql:Context` can be used to invalidate caches in a `graphql:Service`. To invalidate a specific cache in a `resource` function, the `invalidate()` API requires to specify the name of the resource. On the other hand, the `invalidateAll()` API invalidates all caches within the service.

Reviewer: We need to explain how to provide the path when we have a field deeper in the graph. @DimuthuMadushan Can you check this?

Use `resource method` instead of `resource function`.

> The Ballerina `graphql` module provides functionality for cache invalidation. The `invalidate(resourcePath: string)` and `invalidateAll()` APIs in the `graphql:Context` can be used to invalidate caches in a `graphql:Service`. To invalidate a specific cache in a `resource` function, the `invalidate(resourcePath: string)` API requires to specify the name of the resource. On the other hand, the `invalidateAll()` API invalidates all caches within the service.

Reviewer: `invalidate(resourcePath: string)` doesn't make sense. It is a Ballerina API, but doesn't look like one here.

> The Ballerina `graphql` module provides functionality for cache invalidation. The `evictCache(resourcePath: string)` and `invalidateAll()` APIs in the `graphql:Context` can be used to invalidate caches in a `graphql:Service`. To invalidate a specific cache in a `resource` function, the `evictCache(resourcePath: string)` API requires the developer to specify the name of the resource. On the other hand, the `invalidateAll()` API invalidates all caches within the service.

Reviewer 2: ```suggestion
The Ballerina `graphql` module provides functionality for cache invalidation. The `invalidate(resourcePath: string)` and `invalidateAll()` APIs in the `graphql:Context` can be used to invalidate caches in a GraphQL service. To invalidate a specific cache in a `resource` function, the `invalidate(resourcePath: string)` API requires the developer to specify the name of the resource. On the other hand, the `invalidateAll()` API invalidates all caches within the service.
```
We have renamed the `evictCache()` to `invalidate()`. Check other places as well.

Author: Fixed in [1278ea3](https://github.com/ballerina-platform/ballerina-distribution/pull/5188/commits/1278ea379a55bf994c3f7d6de626a2bf60aba971)

> The Ballerina `graphql` module provides functionality for cache invalidation. The `invalidate()` and `invalidateAll()` APIs in the `graphql:Context` can be used to invalidate caches in a `graphql:Service`. To invalidate a specific cache in a `resource` method, the `invalidate()` API requires to specify the name of the resource method. On the other hand, the `invalidateAll()` API invalidates all caches within the service. Additionally, the `invalidate()` API supports the cache invalidation of specific subfields, as exemplified by the syntax `invalidate("field.sub-field")`. Conversely, the `invalidateAll()` API invalidates all caches within the `graphql:Service`.

Reviewer: This still isn't accurate. :( The `invalidate` method is for invalidating the cache of a specific field. It's not an additional feature, it is the feature. 

```suggestion
The Ballerina `graphql` module provides functionality for cache invalidation. The `invalidate()` and `invalidateAll()` APIs in the `graphql:Context` can be used to invalidate caches in a `graphql:Service`. The `invalidate()` API supports the cache invalidation of a specific field by providing the full path of the field separated by a full stop(`.`). For example, `invalidate("field.subfield.anotherSubfield")`. Conversely, the `invalidateAll()` API invalidates all caches within the `graphql:Service`.. On the other hand, the `invalidateAll()` API invalidates all caches within the service.
```
